docs/developers/config_tin.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+from enum import Enum
+
+
+class ArmoryMode(Enum):
+    HOST = "host"
+    DOCKER = "docker"
+    OFFLINE = "offline"
+
+
+@dataclass
+class ArmoryCredentials:
+    github_token: str
+    s3_id: str
+    s3_key: str
+
+
+@dataclass
+class ArmoryPaths:
+    output_dir: Path
+    local_git_dir: Path
+
+
+def paths_by_mode(mode: str) -> ArmoryPaths:
+    if mode == "docker":
+        return ArmoryPaths(output_dir=Path("/output"), local_git_dir=Path("/git"))
+    elif mode == "host":
+        return ArmoryPaths(
+            output_dir=Path("/tmp/armory"), local_git_dir=Path("/tmp/armory-local")
+        )
+    else:
+        raise ValueError(f"Unknown mode: {mode}")
+
+
+@dataclass
+class ArmoryFlags:
```

Comment:
I think these should be broken apart, and some should be moved to `ExperimentParameters`.  For example:

`interactive` is only meaningful in docker mode so maybe we have something like `DockerOptions`

`jupyter` is a bit fuzzy to me, but I think it only works in docker mode (correct me if I am wrong)
`host_port` seems to only apply to `jupyter`.... so we should combine these to something like `jupyter_port` which could be None if `jupyter` is off, or the port if it is enabled.

`check_run`, `num_eval_batches`, `skip_benign`, `skip_attack`, `skip_misclassified`, `validate_config` are all `ExperimentParameters` so should be moved to that object (which should be in `armory.experiment`). Also, I don't know what `validate_config` means but why would you ever not do that?

Reply:
Per our discussion, this is an open question.  We need better definition generally.